compiler/src/dotty/tools/dotc/transform/MixinOps.scala

```diff
@@ -59,10 +64,14 @@ class MixinOps(cls: ClassSymbol, thisTransform: DenotTransformer)(implicit ctx:
     def needsDisambiguation = competingMethods.exists(x=> !(x is Deferred)) // multiple implementations are available
     def hasNonInterfaceDefinition = competingMethods.exists(!_.owner.is(Trait)) // there is a definition originating from class
     meth.is(Method, butNot = PrivateOrAccessorOrDeferred) &&
-    (meth.owner.is(Scala2x) || needsDisambiguation || hasNonInterfaceDefinition ) &&
+    (meth.owner.is(Scala2x) || needsDisambiguation || hasNonInterfaceDefinition || needsJUnit4Fix(meth) ) &&
     isCurrent(meth)
   }
 
+  private def needsJUnit4Fix(meth: Symbol): Boolean = {
+    meth.annotations.nonEmpty && JUnit4Annotations.exists(annot => annot.exists && meth.hasAnnotation(annot))
```

Comment:
Why `annot.exists` if they are already filtered out in [`filter(_.exists)`](https://github.com/lampepfl/dotty/pull/2557/files#diff-18af2904c028f762167dfeae745bb6ceR19)?